opensrp-chw/src/androidTest/java/org/smartregister/chw/activity/ba/ANCRegistrationTestsBa.java

```diff
@@ -0,0 +1,152 @@
+
+    @Rule
+    public GrantPermissionRule mRuntimePermissionRule = GrantPermissionRule.grant(Manifest.permission.CAMERA);
+
+    @Rule
+    public GrantPermissionRule mRuntimePermissionRule1 = GrantPermissionRule.grant(Manifest.permission.ACCESS_FINE_LOCATION);
+
+    private Utils utils = new Utils();
+
+    public void setUp() throws InterruptedException {
+        Thread.sleep(10000);
+        utils.logIn(Constants.BoreshaAfyaConfigUtils.ba_username, Constants.BoreshaAfyaConfigUtils.ba_password);
+    }
+
+    @Test
+    @Order(order = 2)
+    public void registerANCSuccessfuly() throws Throwable {
+        onView(ViewMatchers.withSubstring(Configs.TestConfigHelper.familyName + " Family"))
+                .perform(click());
+        /*onView(androidx.test.espresso.matcher.ViewMatchers.withSubstring(Configs.TestConfigs.aboveFiveFirstNameTwo
+                + " " + Configs.TestConfigs.aboveFiveSecondNameTwo + " " + Configs.TestConfigs.familyName + ", "
+                + Configs.TestConfigs.aboveFiveage))
+                .perform(click());*/
+        utils.locateLayout(1,2).perform(click());
+        Thread.sleep(500);
+        utils.openFamilyMenu();
+        Thread.sleep(500);
+        onView(ViewMatchers.withSubstring("ANC Registration"))
+                .perform(click());
+        Activity activity = getCurrentActivity();
+        Thread.sleep(500);
+        onView(withId(getViewId((JsonFormActivity) activity, "step1:last_menstrual_period_unknown")))
+                .perform(click());
+        onView(withId(getViewId((JsonFormActivity) activity, "step1:edd")))
+                .perform(click());
+        Thread.sleep(1000);
+        onView(ViewMatchers.withSubstring("done")).perform(click());
+        Thread.sleep(500);
+        onView(withId(getViewId((JsonFormActivity) activity, "step1:no_prev_preg")))
+                .perform(typeText("0"));
+        Thread.sleep(500);
+        onView(withId(getViewId((JsonFormActivity) activity, "step1:phone_number")))
+                .perform(scrollTo(), clearText(), typeText("0882454545"));
+        Thread.sleep(500);
+        onView(withId(getViewId((JsonFormActivity) activity, "step1:marital_status")))
+                .perform(scrollTo(), click());
+        onView(ViewMatchers.withSubstring("Co-habiting"))
+                .perform(click());
+        Thread.sleep(500);
+        onView(withId(getViewId((JsonFormActivity) activity, "step1:person_assist")))
+                .perform(scrollTo(), click());
+        onView(ViewMatchers.withSubstring("No"))
+                .perform(click());
+        Thread.sleep(500);
+        onView(ViewMatchers.withSubstring("Save"))
+                .perform(click());
+        onView(ViewMatchers.withSubstring("ANC Clients"))
+                .check(matches(isDisplayed()));
+    }
+
+    @Test
+    @Order(order = 1)
+    public void registerANCWithoutAllFields() throws Throwable {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5426621)